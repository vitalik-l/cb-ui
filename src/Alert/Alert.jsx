import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

class Alert extends PureComponent {
    static propTypes = {
        type: PropTypes.string,
        content: PropTypes.any
    };

    render() {
        const {type, content} = this.props;
        return (
            <div className={classNames('alert', type, {active: !!content})}>
                {content}
            </div>
        );
    }
}

export default Alert;