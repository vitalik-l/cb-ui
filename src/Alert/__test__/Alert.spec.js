import React from 'react';
import {shallow} from 'enzyme';
import Alert from '../Alert';

describe('Alert', () => {
   const wrapper = shallow(<Alert />);

   it('non active', () => {
      expect(wrapper).toMatchSnapshot();
   });

   it('active', () => {
      wrapper.setProps({content: 'test'});
      expect(wrapper).toMatchSnapshot();
   });
});