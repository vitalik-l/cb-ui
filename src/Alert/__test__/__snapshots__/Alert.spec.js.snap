// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert active 1`] = `
<div
  className="alert active"
>
  test
</div>
`;

exports[`Alert non active 1`] = `
<div
  className="alert"
/>
`;
